paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/GooglePayState.kt

```diff
@@ -0,0 +1,14 @@
+package com.stripe.android.paymentsheet.state
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+internal sealed class GooglePayState : Parcelable {
+    object Available : GooglePayState()
+    object NotAvailable : GooglePayState()
+    object Indeterminate : GooglePayState()
+
+    val isReadyForUse: Boolean
```

Comment:
Another way to do this is to pass a parameter into the `GooglePayState` constructor. Then you don't have to rely on a reflection call to see if it's the right thing.